TLM/TLM/Manager/Impl/VehicleBehaviorManager.cs

```diff
@@ -1714,11 +1714,12 @@ IJunctionRestrictionsManager junctionRestrictionsManager
             return checkSpace;
         }
 
-        public bool MayDespawn(ref Vehicle vehicleData) {
+        public bool MayDespawn(ushort vehicleId, ref Vehicle vehicleData) {
             return !Options.disableDespawning
                    || ((vehicleData.m_flags2 & (Vehicle.Flags2.Blown
                                                 | Vehicle.Flags2.Floating)) != 0)
-                   || (vehicleData.m_flags & Vehicle.Flags.Parking) != 0;
+                   || (vehicleData.m_flags & Vehicle.Flags.Parking) != 0
+                   || GlobalConfig.Instance.Gameplay.AllowedDespawnVehicleTypes.IsFlagSet(ExtVehicleManager.Instance.ExtVehicles[vehicleId].vehicleType);
```

Comment:
This uses `ColossalFramework.EnumExtensions.IsFlagSet<T>` which is slow and creates garbage specially in a loop.
Is this performance critical

Reply:
If you hover over it and you see `<type>` then it should raise the alarm :)
![image](https://user-images.githubusercontent.com/26344691/156879166-580518e6-c5ba-4bde-8d0e-6a856f84465e.png)

Reply:
Hmm, I see there are more in the project 😕

Reply:
I'll create new PR, I'm searching if we could create Roslyn analyzer to warn or throw error when certain types are used when compiling the project. It looks like it's possible and not that hard.

Reply:
But none-of them are performance critical except maybe the one here. BTW is this method performance critical? If not then this is OK (maybe put a comment that this is not performance critical - I wish Jetbrains had annotations for this.)
![image](https://user-images.githubusercontent.com/26344691/156892646-7f6b2aac-a03f-41e7-b1ad-71b4d5fd8332.png)

Reply:
I checked a little bit and I can see that this method is performance critical as it is used in simulation step of vehicle AI.

Reply:
Yeah besides UI which can be ignored we have this:
![image](https://user-images.githubusercontent.com/19638970/156893087-ca0dcf07-29c5-4048-ba67-5dbdf0909839.png)

Reply:
We need to do some performance benchmarks from time to time. stuff like this can sneak in.

My AN mod had a similar problem (which went unnoticed for a year somehow) dropping FPS from 90 to 20!
https://github.com/kianzarrin/AdaptiveNetworks/commit/03e6b714800f5aa3b58d567babf6999ee3d558df

Reply:
Yeah, or... create static code analyzer extension with list of banned namespaces/types.
Benchmark won't really help if you don't know where the problem is (assuming you noticed there is a problem)

Reply:
@krzychu124 PAY ATTENTION to this https://github.com/CitiesSkylinesMods/TMPE/pull/1441#discussion_r820079475!
Not every IsFlagSet is bad. Only the generic version is slow. 
![image](https://user-images.githubusercontent.com/26344691/156893215-d0e3d60e-548e-4ee2-a23b-427397a0afc8.png)

It would be wrong to search for `IsFlagSet`. you need to find references to `IsFlagSet<>` only. 

Like this:
![image](https://user-images.githubusercontent.com/26344691/156893440-7a720ede-5b98-44c5-95ee-7c562aa1638d.png)

Reply:
@DaEgi01 was complaining `isFlagSet` can be confused like this.

Reply:
First: 
![image](https://user-images.githubusercontent.com/19638970/156893546-874e56dd-14fc-471b-92c5-f5c869299dd1.png)

Then:
![image](https://user-images.githubusercontent.com/19638970/156893559-8ecd8844-c1c2-4a26-95e2-21f313c0cf71.png)

The problem is... I see something completely different. It shows `CO.EnumExtensions`. Sometimes I feel like it is random...
The only namespace imported is `ColossalFramework`. What a mess.

Reply:
It shows correct extension when I move `using ColossalFramework;` outside of the namespace declaration...
@DaEgi01 how this thing works? Which is more important?

Reply:
> It shows correct extension when I move using ColossalFramework;

no way. there is not fast `IsFlagSet` for ExtVehicleType 
![image](https://user-images.githubusercontent.com/26344691/156894212-a09b4792-4861-471b-b1e4-cfb8f289312a.png)

Reply:
>  Sometimes I feel like it is random

Maybe there is a problem with kyder. VS does not have that problem.

Reply:
I was talking about CitizenInstance IsFlagSet (your first screenshot), there is no IsFlagSet for ExtVehicleType, obviously.

Reply:
In that case maybe egi was right to rename `IsFlagsSet()` to `IsAnyFlagSet` or something.

Reply:
we could also use static method call like `FlagExtensions.IsFlagSet(currentValue, valueToTest)`

Reply:
What if we rename our `IsFlagSet` extensions to something like `HasFlag` to clearly differentiate from the slow generic method in CO code?

Reply:
you can already do this with the current extension methods.
But first we need a fast version for ExtVehicleType.